### PR TITLE
[DO NOT MERGE] CI probe: attribute pre-existing failures on 001

### PR DESCRIPTION
**Pull request by Bovey0809**
Empty commit on top of 001-stereo-centernet-gaps HEAD (4d509da41, which was pushed without CI). Purpose: determine whether the Benchmarks/GPU/Docs failures seen on #25421 pre-exist on the base branch. Will be closed after CI completes.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🧹 Modernizes Ultralytics code, improves error handling and data-processing edge cases, and refreshes documentation and examples without changing core model behavior.

### 📊 Key Changes
- Replaced legacy `dict(...)` construction, unnecessary `pass` statements, redundant loops, and older import styles with cleaner modern Python syntax across the codebase.
- Preserved original exception tracebacks by using bare `raise` in export, training, TensorRT, retry, and model-saving paths.
- Improved semantic dataset selection by recognizing datasets with an existing `masks` directory.
- Fixed several edge cases in SAM tracking, stereo 3D visualization, dataset conversion, class-weight handling, and default mutable arguments.
- Updated type annotations, including progress-bar context managers and batch-size utilities.
- Refreshed tests, documentation generation, plotting references, and the object-tracking notebook using YOLO26 segmentation.
- Simplified callback, validator, and trainer no-op methods while maintaining their extension points.
- Improved file downloading, tuner synchronization, version checks, source metadata assignment, and Git branch parsing.

### 🎯 Purpose & Impact
- ✅ Makes the codebase more consistent, readable, and maintainable for contributors.
- 🐞 Reduces subtle runtime issues by preserving traceback context and handling edge cases more safely.
- 📦 Improves reliability for semantic segmentation, SAM workflows, tracking, 3D detection, and model export.
- 🧪 Keeps CLI, Python, CUDA, and export workflows covered by updated tests.
- 📚 Provides clearer documentation and a more current YOLO26 tracking example for users.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Bovey0809, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. An automated review has been added, and an Ultralytics engineer will assist with the next steps. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The diff is largely mechanical cleanup. Three regressions need attention: Python 3.8 compatibility, an undeclared/insufficient typing dependency on the unconditional import path, and a newly unsuppressed Ruff E731 violation.

💬 Posted 3 inline comments

File: ultralytics/utils/export/qnn.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def onnx2qnn(
     dims = [d.dim_value for d in onnx.load(str(onnx_file)).graph.input[0].type.tensor_type.shape.dim]
     if len(dims) == 4 and dims[3] in {1, 3} and dims[1] not in {1, 3}:  # channel-last graph (QNNModel export)
         nchw_transform = transform_fn
-        transform_fn = lambda data_item: nchw_transform(data_item).transpose(0, 2, 3, 1)  # noqa: E731
+        transform_fn = lambda data_item: nchw_transform(data_item).transpose(0, 2, 3, 1)
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/export/qnn.py:99]: 💡 **MEDIUM**: Removing the existing `# noqa: E731` leaves a lambda assignment that Ruff's default E7 rules reject, so the lint job will fail. Keep the suppression or replace this with a named local function.

File: ultralytics/utils/git.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def branch(self) -> str | None:
         if not self.is_repo or not self.head or not self.head.startswith("ref: "):
             return None
         ref = self.head[5:].strip()
-        return ref[len("refs/heads/") :] if ref.startswith("refs/heads/") else ref
+        return ref.removeprefix("refs/heads/")
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/git.py:130]: 💡 **MEDIUM**: `str.removeprefix()` was added in Python 3.9, but the package explicitly supports Python 3.8. Accessing `GitRepo.branch` on Python 3.8 will raise `AttributeError`. Retain the prefix-slice implementation (or an equivalent compatible form).

**Suggested change:**
```suggestion
        return ref[len("refs/heads/") :] if ref.startswith("refs/heads/") else ref
```

File: ultralytics/utils/tqdm.py [hunk 3]
```diff
@@ -8,6 +8,8 @@
 from functools import lru_cache
 from typing import IO, Any
 
+from typing_extensions import Self
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/tqdm.py:11]: 💡 **MEDIUM**: `typing_extensions` is not a declared direct dependency, and `Self` is unavailable in older `typing_extensions` versions that remain valid with the supported `torch>=1.8` floor. Since this module is imported unconditionally by `ultralytics.utils`, affected environments will fail during `import ultralytics`. Avoid the new import by annotating `__enter__` with `TQDM` (postponed annotations are already enabled), or declare a compatible minimum `typing_extensions` dependency.

**Comment by Bovey0809** [comment]
Probe complete: bare 001 HEAD (4d509da41, pushed without CI) fails the identical job set as #25421 — Benchmarks x3, Docs, IsolatedExports, and all Tests jobs (s3d predict/val/export tests with the released yolo26n-s3d.pt); GPU nan-recovery crash is intermittent base code (trainer.py int(None)). The failures pre-date #25421. The base branch needs its own fix.
